app/src/main/java/org/simple/clinic/overdue/AppointmentRepository.kt

```diff
@@ -84,7 +86,7 @@ class AppointmentRepository @Inject constructor(
     return Completable.fromAction {
       appointmentDao.markAsAgreedToVisit(
           appointmentUUID = appointmentUuid,
-          reminderDate = LocalDate.now(utcClock).plusDays(30),
+          reminderDate = LocalDate.now(userClock).plusDays(30),
```

Comment:
Can you get the `clock` from the caller itself rather than injecting in the repository?

Reply:
Yup. 

Following up on our conversation offline, it makes sense to move this changing of reminder date to the controller and let the repository handle only setting the record timestamps (`createdAt`, `updatedAt`, and `deletedAt`). I have raised a pivotal story for this [here](https://www.pivotaltracker.com/story/show/168449411) and I'll get to it at some point.